aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -888,6 +896,119 @@ module aptos_framework::delegation_pool {
         }
     }
 
+    /// Initialize a delegation pool from an existing owned staking contract.
+    /// The newly created delegation pool will preserve the operator and commission fee. The staker becomes the pool
+    /// owner and has the same delegated voter set.
+    public entry fun initialize_delegation_pool_from_staking_contract(
+        staker: &signer,
+        operator: address,
+    ) acquires DelegationPool, GovernanceRecords, BeneficiaryForOperator, NextCommissionPercentage {
+        assert!(features::delegation_pools_enabled(), error::invalid_state(EDELEGATION_POOLS_DISABLED));
+        assert!(
+            features::staking_contract_to_delegation_pool_conversion_enabled(),
+            error::invalid_state(ESTAKING_CONTRACT_TO_DELEGATION_POOL_CONVERSION_NOT_SUPPORTED)
+        );
+
+        let staker_address = signer::address_of(staker);
+        assert!(!owner_cap_exists(staker_address), error::already_exists(EOWNER_CAP_ALREADY_EXISTS));
+
+        // do not have to withhold the `add_stake` fee from staker if there is no pending_active stake
+        let (_, _, pending_active, _) = stake::get_stake(
+            staking_contract::stake_pool_address(staker_address, operator)
+        );
+        assert!(pending_active == 0, error::invalid_state(EPENDING_ACTIVE_STAKE_ON_STAKING_CONTRACT));
+
+        // destroy staking contract and take ownership of its underlying stake pool
+        let (
+            principal,
+            pool_address,
+            owner_cap,
+            operator_commission_percentage,
+            distribution_pool,
+            stake_pool_signer_cap,
+        ) = staking_contract::destroy_staking_contract(staker, operator);
+
+        // convert existing commission to a 2-decimals precision percentage
+        operator_commission_percentage = operator_commission_percentage * 100;
+
+        let stake_pool_signer = account::create_signer_with_capability(&stake_pool_signer_cap);
+        coin::register<AptosCoin>(&stake_pool_signer);
+
+        // stake_pool_signer becomes owner of the stake pool and stores its `stake::OwnerCapability`
+        stake::deposit_owner_cap(&stake_pool_signer, owner_cap);
+
+        let inactive_shares = table::new<ObservedLockupCycle, pool_u64::Pool>();
+        table::add(
+            &mut inactive_shares,
+            olc_with_index(0),
+            pool_u64::create_with_scaling_factor(SHARES_SCALING_FACTOR)
+        );
+
+        let pool = DelegationPool {
+            active_shares: pool_u64::create_with_scaling_factor(SHARES_SCALING_FACTOR),
+            observed_lockup_cycle: olc_with_index(0),
+            inactive_shares,
+            pending_withdrawals: table::new<address, ObservedLockupCycle>(),
+            stake_pool_signer_cap,
+            // `staking_contract::distribute_internal` has already withdrawn any existing inactive stake
```

Comment:
I wonder if we should just do an assert that inactive == 0 just to be absolutely certain

Reply:
should be use the `error::internal` code as this is an invariant not expected to fail? we can also formally prove it